activesupport/lib/active_support/testing/setup_and_teardown.rb

```diff
@@ -36,12 +36,12 @@ def teardown(*args, &block)
         end
       end
 
-      def before_setup # :nodoc:
+      def setup # :nodoc:
```

Comment:
This change can't be done. It is breaking a lot of tests, as you can see for all the files you had to change. The commit message for this change also is wrong. It say we can't use those hooks, but the hook documentation says clearly: `This hook is meant for libraries to extend minitest.` We are a library and we are extending minitest

Reply:
Thanks for the review! I think you're right, I didn't realize how much of a breaking change that would be.

I believe I have a more elegant solution that shouldn't break anything that is ready for review!

There are two test thread failures in `actionpack` that seem to be flakey tests, and not related to my changes.